inference/core/workflows/core_steps/common/utils.py

```diff
@@ -93,10 +129,11 @@ def anchor_prediction_detections_in_parent_coordinates(
 
 def anchor_detections_in_parent_coordinates(
```

Comment:
[loud thinking] what happens with parent detection when we want to modify a detection? is not that expected that when we shift or filter - the exact same thing happens for parent?
could we even address this with `sv.Detections`? at first, I thought about having it nested somehow, but that seems to be hacky and unclear as the idea. Getting into conclusion that maybe we should just keep `shift_x, shift_y` as metadata of each detection and only execute this op when needed (basically the only need is while output construction) - then the processing gets more light-weight, as we do not need to perform laborious ops each step